### PR TITLE
package/pinniped: post-deploy job read nodes to obtain cluster-name

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Signed-off-by: Benjamin A. Petersen <ben@benjaminapetersen.me>

## What this PR does / why we need it

- adds the ability to `list nodes` to the post-deploy job to each of the following (currently supported) versions of the pinniped-package: 
   - 0.12.1
   - 0.12.0
   - 0.4.4
- Builds a new package image for each of the following    
   - `make push-package PACKAGE=pinniped VERSION=0.12.1 TAG=0.12.1-4-post-deploy-rbac-nodes` pushed `projects.registry.vmware.com/tce/pinniped@sha256:4f7ed9a21171035a390bbaf40cd522a22f82b126602968e2ebba040010eb9393` and updated `0.12.1/package.yaml`
   - `make push-package PACKAGE=pinniped VERSION=0.12.0 TAG=0.12.0-4-post-deploy-rbac-nodes` pushed `projects.registry.vmware.com/tce/pinniped@sha256:349c0c433440e45e8434d9cdf7d87d8536b5be06b515ae0eadac1248be8a2154` and updated `0.12.0/package.yaml`
   - `make push-package PACKAGE=pinniped VERSION=0.4.4 TAG=0.4.4-4-post-deploy-rbac-nodes` pushed `projects.registry.vmware.com/tce/pinniped@sha256:2eb6a2e8a489837b6a50e7661313b25ad5d07ade3246456c268ad3ae537c76dd` and updated `0.4.4/package.yaml`
- Updates the .imgpkg/images.yml file in each of the packages via `ytt -f bundle/config/ -f bundle/examples/mc-ldap.yaml | kbld -f bundle/kbld-config.yaml -f - --imgpkg-lock-output bundle/.imgpkg/images.yml`: 
   - 0.12.1
   - 0.12.0 (ran cmd but no change to current package)
   - 0.4.4

### Why do this for the currently supported 3 package versions?

The `tkg-metadata` configmap may not always be available in future versions of TKG.  If so, the post-deploy job can read an annotation off of a `node` to get the `cluster-name`.  The annotation is `"cluster.x-k8s.io/cluster-name"`.  The paper trail for this `cluster-api` feature can be [ found here.](https://github.com/kubernetes-sigs/cluster-api/issues/4044)

## Which issue(s) this PR fixes

Fixes: #

## Describe testing done for PR

- Manually swapped the pinniped package on a running cluster and verified the new RBAC rules exist
- Manually built a new post-deploy job and, via overlay, swapped an image using the `list nodes` ability and verified the annotation can be read.

### CODE REVIEW

File: addons/packages/pinniped/0.12.1/bundle/config/upstream/03-post-deploy.yaml [hunk 1]
```diff
@@ -61,6 +63,7 @@ metadata:
   name: #@ post_deploy_job_name()
   namespace: #@ pinniped_supervisor_namespace()
 spec:
+  backoffLimit: 10 #! should we increase from the default 4?  No reason to leave this such a low number at this point...
```
[the remark this answers is not in the document]
Comment (benjaminapetersen) [reply addons/packages/pinniped/0.12.1/bundle/config/upstream/03-post-deploy.yaml:66]: Mostly a reminder to myself atm, with review will discuss removal or split off into separate PR.

**Review by ankeesler — APPROVED** [review]

**Comment by benjaminapetersen** [comment]
Will circle back and update all the unit tests shortly.

**Comment by benjaminapetersen** [comment]
Will circle back and update all the unit tests shortly.

File: addons/packages/pinniped/0.12.1/bundle/config/upstream/03-post-deploy.yaml [hunk 2]
```diff
@@ -61,6 +63,7 @@ metadata:
   name: #@ post_deploy_job_name()
   namespace: #@ pinniped_supervisor_namespace()
 spec:
+  backoffLimit: 10  ### THIS THING HERE
```
Comment (benjaminapetersen) [inline addons/packages/pinniped/0.12.1/bundle/config/upstream/03-post-deploy.yaml:66]: remove this comment.

File: addons/packages/pinniped/0.12.1/bundle/examples/mc-idp-type-ninnyhammer.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#@data/values
```
Comment (benjaminapetersen) [inline addons/packages/pinniped/0.12.1/bundle/examples/mc-idp-type-ninnyhammer.yaml:1]: Remove this file, its not a real "example".

File: addons/packages/pinniped/0.12.1/bundle/examples/wlc-idp-type-ninnyhammer.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#@data/values
```
Comment (benjaminapetersen) [inline addons/packages/pinniped/0.12.1/bundle/examples/wlc-idp-type-ninnyhammer.yaml:1]: Remove this file, its not a real "example".

File: addons/packages/pinniped/0.12.1/bundle/config/upstream/03-post-deploy.yaml (hunk 2 above)
Comment (benjaminapetersen) [reply addons/packages/pinniped/0.12.1/bundle/config/upstream/03-post-deploy.yaml]: done.

File: addons/packages/pinniped/0.12.1/bundle/examples/wlc-idp-type-ninnyhammer.yaml (hunk 4 above)
Comment (benjaminapetersen) [reply addons/packages/pinniped/0.12.1/bundle/examples/wlc-idp-type-ninnyhammer.yaml]: done.

File: addons/packages/pinniped/0.12.1/bundle/examples/mc-idp-type-ninnyhammer.yaml (hunk 3 above)
Comment (benjaminapetersen) [reply addons/packages/pinniped/0.12.1/bundle/examples/mc-idp-type-ninnyhammer.yaml]: done.

**Comment by benjaminapetersen** [comment]
Closing this for now, we don't need the ability to read nodes yet (but likely will in the future).

**Pull request closed without merging**